internal/service/certificatesigningrequest.go

```diff
@@ -8,13 +8,71 @@ import (
 
 	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/crypto"
 	"github.com/flightctl/flightctl/internal/flterrors"
 	"github.com/flightctl/flightctl/internal/service/common"
 	"github.com/flightctl/flightctl/internal/store"
 	"github.com/go-openapi/swag"
+	"github.com/google/uuid"
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+const DefaultEnrollmentCertExpirySeconds int32 = 60 * 60 * 24 * 7 // 7 days
+
+func signApprovedCertificateSigningRequest(ca *crypto.CA, request *api.CertificateSigningRequest) error {
+	if request == nil {
+		return errors.New("signApprovedCertificateSigningRequest: CertificateSigningRequest is nil")
+	}
+
+	if request.Metadata.Name == nil {
+		return fmt.Errorf("signApprovedCertificateSigningRequest: certificate signing request is missing metadata.name")
+	}
+
+	// NOTE: this currently only requires one approval; this will change in the future based on policy
+	// TODO: do we need CertificateSigningRequestApproved instead of the string?
+	approved := false
+	for _, c := range request.Status.Conditions {
+		if c.Type == "Denied" {
```

Comment:
This just means that the "Denied" condition exists - it doesn't check if the condition is True or False.  So you could be returning an error here for Denied=False.
I suggest using the helper function `api.IsStatusConditionTrue(request.Status.Conditions, api.CertificateSigningRequestDenied)`

(Also note using the type rather than hard-coded string)

Reply:
The helper function is quite handy, thank you